### PR TITLE
fix #871

### LINKED ISSUE #871: Error: spawn /Applications/Browser.html.app/Contents/MacOS/graphene ENOENT
I get `Error: spawn /Applications/Browser.html.app/Contents/MacOS/graphene ENOENT` that error on linux.

To fix on gulpfile.babel.js on line 24 I changed with my path of graphene.
Maybe add some info about that in readme?

### PR DESCRIPTION
**Pull request by Mte90**

### CODE REVIEW

File: gulpfile.babel.js [hunk 1]
```diff
@@ -44,6 +45,13 @@ var settings = {
   compression: null
 };
 
+fs.exists(settings.geckoPath, function (exists) {
+  if(!exists) {
+    console.log("Error: Graphene on " + settings.geckoPath + " not found!");
+    process.exit(1);
+  } 
+});
+
```
Comment (paulrouget) [inline gulpfile.babel.js]: Don't you want to add this check in the gecko task?

Comment (Mte90) [inline gulpfile.babel.js]: done

**Comment by Mte90** [comment]
updated and merge as 1 commit

**Comment by paulrouget** [comment]
Use this error message: `"Error: Gecko binary not found: " + settings.geckoPath`;

Can you do the same thing for Servo (the default location should be `/usr/local/bin/servo`).

Ideally, it would use the OS path resolution (using the `PATH` env variable), but that can be done later.

Thank you!

**Comment by Mte90** [comment]
done also for servo and merged in 1 commit

File: gulpfile.babel.js [hunk 2]
```diff
@@ -116,41 +117,54 @@ gulp.task('server', function() {
...
 gulp.task('servo', function() {
-  var app = child.spawn(settings.servoPath, [
-    '-w',
-    '-b',
-    '--pref',
-    'dom.mozbrowser.enabled',
-    'http://localhost:' + settings.port
-  ], {
-    stdio: 'inherit'
+  fs.exists(settings.geckoPath, function (exists) {
```
Comment (paulrouget) [inline gulpfile.babel.js]: you want to check servoPath, not geckoPath.

File: gulpfile.babel.js [hunk 3]
```diff
@@ -116,41 +117,54 @@ gulp.task('server', function() {
...
+    if(exists) {
+      var app = child.spawn(settings.servoPath, [
+          '-w',
+            '-b',
+            '--pref',
+            'dom.mozbrowser.enabled',
+            'http://localhost:' + settings.port
```
Comment (paulrouget) [inline gulpfile.babel.js]: indentation issue.

File: gulpfile.babel.js (hunk 2 above)
Comment (paulrouget) [inline gulpfile.babel.js]: Also, add a default path to `servoPath: process.env.BROWSER_HTML_SERVO_PATH`, like it is done for geckoPath. (`|| /usr/local/bin/servo`)

**Comment by paulrouget** [comment]
That was fast :)

And the update cut this comment out:

"Also, add a default path to servoPath: process.env.BROWSER_HTML_SERVO_PATH, like it is done for geckoPath. (|| `/usr/local/bin/servo`)"

**Comment by Mte90** [comment]
Fixed everything and added the default path for servo.

File: gulpfile.babel.js [hunk 4]
```diff
@@ -16,13 +16,15 @@ import sequencial from 'gulp-sequence';
...
 import hmr from 'browserify-hmr';
 import hotify from 'hotify';
+var fs = require('fs');
 
 var settings = {
   port: process.env.BROWSER_HTML_PORT ||
         '6060',
-  geckoPath: process.env.BROWSER_HTML_GECKO_PATH ||
+  geckoPath: process.env.BROWSER_HTML_GECKO_PATH |
         '/Applications/Browser.html.app/Contents/MacOS/graphene',
```
Comment (paulrouget) [inline gulpfile.babel.js]: `||`, not `|`

**Comment by Mte90** [comment]
ops i hope that now is ok

File: gulpfile.babel.js [hunk 5]
```diff
@@ -116,41 +118,54 @@ gulp.task('server', function() {
...
+      });
+      var exit = function(code) {
+          app.kill();
+          process.exit(code);
+      }
+
+      process.on('SIGINT', exit);
+      app.on('close', exit);
+    } else {
+      console.log("Error: Gecko binary not found: " + settings.geckoPath;);
```
Comment (paulrouget) [inline gulpfile.babel.js]: extra `;`

File: gulpfile.babel.js [hunk 6]
```diff
@@ -116,41 +118,54 @@ gulp.task('server', function() {
...
+
+      var exit = function(code) {
+          app.kill();
+          process.exit(code);
+      }
+
+      process.on('SIGINT', exit);
+      app.on('close', exit);
+    } else {
+      console.log("Error: Servo binary not found: " + settings.servoPath;);
```
Comment (paulrouget) [inline gulpfile.babel.js]: extra `;`

Comment (paulrouget) [inline gulpfile.babel.js]: use `console.error`

File: gulpfile.babel.js (hunk 5 above)
Comment (paulrouget) [inline gulpfile.babel.js]: use `console.error`.

**Comment by Mte90** [comment]
My editor not support es6 but i have fixed that bugs.

**Comment by paulrouget** [comment]
Thank you @Mte90 !

**Pull request merged**